lib/ansible/modules/cloud/vmware/vmware_guest_file_operation.py

```diff
@@ -161,240 +161,259 @@
+                                      uuid=self.vm.summary.config.uuid)
+            # other exceptions
+            except Exception as e:
+                self.module.fail_json(msg="Failed to Create directory into VM VMware exception : %s" % to_native(e),
+                                      uuid=self.vm.summary.config.uuid)
+
+        if operation == "delete":
+            try:
+                file_manager.DeleteDirectoryInGuest(vm=self.vm, auth=creds, directoryPath=path,
+                                                    recursive=recurse)
+            except vim.fault.FileNotFound as file_not_found:
+                result['changed'] = False
+                result['msg'] = "Guest directory %s not exists %s" % (path,
+                                                                      to_native(file_not_found.msg))
+            except vim.fault.FileFault as e:
+                self.module.fail_json(msg="FileFault : %s" % e.msg,
+                                      uuid=self.vm.summary.config.uuid)
+            except vim.fault.GuestPermissionDenied as permission_denied:
+                self.module.fail_json(msg="Permission denied for path %s : %s" % (path,
+                                                                                  to_native(permission_denied.msg)),
+                                      uuid=self.vm.summary.config.uuid)
+            except vim.fault.InvalidGuestLogin as invalid_guest_login:
+                self.module.fail_json(msg="Invalid guest login for user %s : %s" % (vm_username,
+                                                                                    to_native(invalid_guest_login.msg)),
+                                      uuid=self.vm.summary.config.uuid)
+            # other exceptions
+            except Exception as e:
+                self.module.fail_json(msg="Failed to Delete directory into Vm VMware exception : %s" % to_native(e),
+                                      uuid=self.vm.summary.config.uuid)
+
+        return result
+
+    def fetch(self):
+        result = dict(changed=True, uuid=self.vm.summary.config.uuid)
+        vm_username = self.module.params['vm_username']
+        vm_password = self.module.params['vm_password']
+        dest = self.module.params["fetch"]['dest']
+        src = self.module.params['fetch']['src']
+        creds = vim.vm.guest.NamePasswordAuthentication(username=vm_username, password=vm_password)
+        file_manager = self.content.guestOperationsManager.fileManager
+
+        try:
+            fileTransferInfo = file_manager.InitiateFileTransferFromGuest(vm=self.vm, auth=creds,
+                                                                          guestFilePath=src)
+            url = fileTransferInfo.url
+            resp, info = urls.fetch_url(self.module, url, method="GET")
+            try:
+                with open(dest, "wb") as local_file:
+                    local_file.write(resp.read())
```

Comment:
Maybe we can write the file in tmp and make a diff or compute it's md5sum to know if the file was already fetched.

But it could be done later.